rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/condition/JavaClass.java

```diff
@@ -330,22 +329,6 @@ public boolean evaluate(Object first, Object second)
     @Override
     public void setParameterStore(ParameterStore store)
     {
-        if (locations != null && !locations.isEmpty())
```

Comment:
It's registered separately in order to separate type interest from the conditions themselves (in order to improve performance). This part may need to get put back to the old way for now if we are going to release this as part of 2.4.